src/pulsar_producer.erl

```diff
@@ -330,6 +414,32 @@ connected(cast, _EventContent, _State) ->
 connected(_EventType, EventContent, State) ->
     handle_response(EventContent, State).
 
+-spec refresh_urls_and_connect(state()) -> handler_result().
+refresh_urls_and_connect(State0) ->
+    %% if Pulsar went down and then restarted later, we must issue a
+    %% LookupTopic command again after reconnecting.
+    %% https://pulsar.apache.org/docs/2.10.x/developing-binary-protocol/#topic-lookup
+    %% > Topic lookup needs to be performed each time a client needs
+    %% > to create or reconnect a producer or a consumer. Lookup is used
+    %% > to discover which particular broker is serving the topic we are
+    %% > about to use.
+    %% Simply looking up the topic (even from a distinct connection)
+    %% will "unblock" the topic so we may send messages to it.  The
+    %% producer may be started only after that.
+    #{ clientid := ClientId
+     , partitiontopic := PartitionTopic
+     } = State0,
+    ?tp(debug, pulsar_producer_refresh_start, #{}),
+    try pulsar_client:lookup_topic_async(ClientId, PartitionTopic) of
```

Comment:
This is essentially the fix for the original bug: after the Pulsar broker is restarted, it requires the topic to be looked up before attempting to send messages to it, even if it already exists.